tests/transformers/test_add_workflow_notification_transformer.py

```diff
@@ -0,0 +1,131 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for AddWorkflowNotificationTransformer"""
+import unittest
+
+from o2a.converter.task import Task
+from o2a.converter.task_group import TaskGroup
+from o2a.converter.workflow import Workflow
+from o2a.o2a_libs.property_utils import PropertySet
+from o2a.transformers.add_workflow_notificaton_transformer import (
+    AddWorkflowNotificationTransformer,
+    PROP_WORKFLOW_NOTIFICATION_URL,
+    START_TASK_GROUP_NAME,
+    END_SUCCESS_TASK_GROUP_NAME,
+    END_FAIL_TASK_GROUP_NAME,
+)
+
+
+class AddWorkflowNotificationTransformerTest(unittest.TestCase):
```

Comment:
nit: Why all the extra empty lines in methods in this class? I personally dislike empty lines, they make the code less readable to me. If we want to group logical sections in tests maybe let's add comment lines like `#given`, `#when` and `#then`? Or we could extract logical sections of code into methods.

Reply:
I updated this PR. I removed extra new lines.

Reply:
I think few of them are still left. Agree with @sprzedwojski that "#Given #When #Then" comments make the code more readable while empty lines do not help with readability. The make you start thinking "what the hell this line is for" . There is a reason for "sparingly" work in https://www.python.org/dev/peps/pep-0008/#blank-lines ("Use blank lines in functions, sparingly, to indicate logical sections."). Given/When/then is much more straightforward ("explicit is better than implicit").

Reply:
I updated this PR.